include/deal.II/grid/grid_tools.h

```diff
@@ -230,6 +230,43 @@ namespace GridTools
   double
   cell_measure(const T &, ...);
 
+  /**
+   * Computes an aspect ratio measure for all cells and fills a vector with one
+   * entry per cell, given a @p triangulation and @p mapping. The aspect ratio
+   * of a cell is defined as the ratio of the maximum to minimum singular value
+   * of the Jacobian, taking the maximum over all quadrature points of a
+   * quadrature rule specified via @p quadrature. For example, for the special
+   * case of rectangular elements in 2d with dimensions $a$ and $b$ ($a \geq
+   * b$), this function returns the usual aspect ratio definition $a/b$. The
+   * above definition using singular values is a generalization to arbitrarily
+   * deformed elements. This function is intended to be used for $d=2,3$ space
+   * dimensions, but it can also be used for $d=1$ returning a value of 1.
+   *
+   * @note Inverted elements do not throw an exception. Instead, a value of inf
+   * is written into the vector in case of inverted elements.
+   *
+   * @note Make sure to use enough quadrature points for a precise calculation
+   * of the aspect ratio in case of deformed elements.
+   *
+   * @author Niklas Fehn, Martin Kronbichler, 2019
+   */
+  template <int dim>
+  Vector<double>
+  calculate_aspect_ratio_of_cells(const Triangulation<dim> &triangulation,
+                                  const Mapping<dim> &      mapping,
+                                  const Quadrature<dim> &   quadrature);
+
+  /**
+   * Computes the maximum aspect ratio by taking the maximum over all cells.
+   *
+   * @author Niklas Fehn, Martin Kronbichler, 2019
+   */
+  template <int dim>
+  double
+  calculate_maximum_aspect_ratio(const Triangulation<dim> &triangulation,
```

Comment:
we call most of these functions compute_*, maybe this is worth changing to be a bit more consistent?

Reply:
changed in https://github.com/dealii/dealii/pull/8894/commits/a55ae94a1579df8590f06e8c7897b0b2401afa61